tests/Collection/CollectionFunctionalTest.php

```diff
@@ -119,6 +120,21 @@ public function testDropIndexShouldNotAllowWildcardCharacter()
         $this->collection->dropIndex('*');
     }
 
+    public function testExplain()
+    {
+        if (version_compare($this->getServerVersion(), '3.0.0', '<')) {
+            $this->markTestSkipped('The explain option requires server version >= 3.0');
+        }
+
+        $this->createFixtures(3);
+
+        $operation = new Count($this->getDatabaseName(), $this->getCollectionName(), ['x' => ['$gte' => 1]], []);
+
+        $result = $this->collection->explain($operation, ['typeMap' => ['root' => 'array', 'document' => 'array']]);
```

Comment:
As mentioned in https://github.com/mongodb/mongo-php-library/pull/500#discussion_r170328237, a custom type map shouldn't be needed here. `assertArrayHasKey()` should be able to handle our BSONDocument as-is.